src/utilities/utilities.ml

```diff
@@ -140,6 +140,9 @@ let rec map3 (f : 'a -> 'b -> 'c -> 'd) l1 l2 l3 : 'd list =
   match (l1, l2, l3) with
   | ([], [], []) ->
      []
+  | ([], _, _) -> failwith "Mismatched sized lists passed to map3"
```

Comment:
These lines were initially defined to address the warning that the pattern matching is not exhaustive. Since I've manually marked this category of warnings to be kept as warnings, this is a bit redundant. I've left this here to give more useful error messages at runtime. Let me know if I should remove them.